pkg/autopilot/controller/updates/update_controller.go

```diff
@@ -34,7 +35,8 @@ type updateController struct {
 
 	clusterID string
 
-	updater *updater
+	updaters  map[string]updater
+	parentCtx context.Context
```

Comment:
Storing [context in a struct](https://go.dev/blog/context-and-structs) is a bit sketchy.  Maybe it can be avoided.

Reply:
This is essentially the "root" signal aware context we have for the whole k0s process. Not sure how to avoid putting that into here as this component also needs to know when to shutdown.

Reply:
Like discussed "offline", we already have couple other places too where the context is stored in structs. We need to figure out a better pattern for this and refactor those few other places too. Let's handle this separate from this PR.

I'll create separate issue to track that.